internal/api/external.go

```diff
@@ -244,12 +259,6 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 		if err := a.setCookieTokens(config, token, false, w); err != nil {
 			return internalServerError("Failed to set JWT cookie. %s", err)
 		}
-	} else {
-		// Left as hash fragment to comply with spec. Additionally, may override existing error query param if set to PKCE.
-		rurl, err = a.prepErrorRedirectURL(unauthorizedError("Unverified email with %v. A confirmation email has been sent to your %v email.", providerType, providerType), w, r, rurl, models.ImplicitFlow)
```

Comment:
This is the surrounding context: https://supabase.slack.com/archives/C022071RB2L/p1686667519790109

Reply:
I couldn't find the section of the spec though @hf it's been a while but do you happen to recall which section of the spec you saw this in?

For the second part of the comment: 
> may override existing error query param if set to PKCE.
I can't recall exactly but I  think if it's set as a query param it might override the error message returned by the OAuth provider. Should have been clearer when I wrote the message  - will check again later in the day

Reply:
I can't seem to find anything here - https://github.com/supabase/gotrue/pull/1129 so I think we should be good to switch to `flowType` instead of pegging it to implicit flow.